pkg/rbd/controllerserver.go

```diff
@@ -111,94 +84,80 @@ func parseVolCreateRequest(req *csi.CreateVolumeRequest) (*rbdVolume, error) {
 	}
 
 	// if it's NOT SINGLE_NODE_WRITER and it's BLOCK we'll set the parameter to ignore the in-use checks
-	rbdVol, err := getRBDVolumeOptions(req.GetParameters(), (isMultiNode && isBlock))
+	rbdVol, err := genVolFromVolumeOptions(req.GetParameters(), (isMultiNode && isBlock))
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	// Generating Volume Name and Volume ID, as according to CSI spec they MUST be different
-	volName := req.GetName()
-	uniqueID := uuid.NewUUID().String()
-	rbdVol.VolName = volName
-	volumeID := "csi-rbd-vol-" + uniqueID
-	rbdVol.VolID = volumeID
+	rbdVol.RequestName = req.GetName()
+
 	// Volume Size - Default is 1 GiB
 	volSizeBytes := int64(oneGB)
 	if req.GetCapacityRange() != nil {
 		volSizeBytes = req.GetCapacityRange().GetRequiredBytes()
 	}
 
-	rbdVol.VolSize = util.RoundUpToMiB(volSizeBytes)
+	// always round up the request size in bytes to the nearest MiB
+	rbdVol.VolSize = util.MiB * util.RoundUpToMiB(volSizeBytes)
 
+	// NOTE: rbdVol does not contain VolID and RbdImageName populated, everything
+	// else is populated post create request parsing
 	return rbdVol, nil
 }
 
-func storeVolumeMetadata(vol *rbdVolume, cp util.CachePersister) error {
-	if err := cp.Create(vol.VolID, vol); err != nil {
-		klog.Errorf("failed to store metadata for volume %s: %v", vol.VolID, err)
-		return err
-	}
-
-	return nil
-}
-
-// CreateVolume creates the volume in backend and store the volume metadata
+// CreateVolume creates the volume in backend
 func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
-
 	if err := cs.validateVolumeReq(req); err != nil {
 		return nil, err
 	}
+
```

Comment:
I dont think this new line is required here.

Reply:
Removed the line space